lib/nero/survey/survey_rules.rb

```diff
@@ -0,0 +1,55 @@
+module Nero
+  module Survey
+    class SurveyRules
+      attr_reader :options
+
+      def initialize(options)
+        @options = options
+      end
+
+      def apply_to(results)
+        return :consensus       if vote_counts(results).any? { |_, count| count >= consensus_limit }
+        return :human           if vote_counts(results)["human"] >= human_limit
+        return :flagged         if vote_counts(results)["reported"] >= flagged_limit
+        return :blank           if votes(results).first(blank_limit).count("blank") == blank_limit
+        return :blank_consensus if vote_counts(results)["blank"] >= blank_consensus_limit
+        false
+      end
+
+      def votes(results)
+        results.map {|result| result["vote"] }
+      end
+
+      def vote_counts(results)
+        votes(results).reduce(Hash.new(0)) do |groups, vote|
```

Comment:
You're right, it'll recalculate this a few times from `apply_to`. The algs this was derived from had a cache, but that broke the test runs (since moving it out of SubjectState, it no longer knows when to clear the cache). I'll move the cache into a local variable in `apply_to`.